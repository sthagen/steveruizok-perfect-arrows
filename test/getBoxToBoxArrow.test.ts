import { getBoxToBoxArrow } from "../src"

describe("getBoxToBoxArrow", () => {
  test("it computes an arrow between boxes", () => {
    expect(getBoxToBoxArrow(10, 10, 10, 10, 30, 30, 10, 10)).toMatchSnapshot()
  })

  test("it computes an arrow from a large box to a small box", () => {
    expect(
      getBoxToBoxArrow(10, 10, 576, 384, 600, 30, 10, 10)
    ).toMatchSnapshot()
  })

  test("it computes an arrow from a box back to itself", () => {
    expect(getBoxToBoxArrow(10, 10, 10, 10, 10, 10, 10, 10)).toMatchSnapshot()
  })

  test("it computes an arrow between boxes with straights disabled", () => {
    expect(
      getBoxToBoxArrow(10, 10, 10, 10, 30, 30, 10, 10, { straights: false })
    ).toMatchSnapshot()
  })

  test("It solves a difficult arrow.", () => {
    expect(
      getBoxToBoxArrow(
        -4710.06155657891,
        -1571.6917781196591,
        545.7397072097903,
        457.4198639994165,
        -5335.457377534619,
        -768.3931456842388,
        559.06478024143,
        358.2843535608206
      )
    ).toBeTruthy()
  })
})
