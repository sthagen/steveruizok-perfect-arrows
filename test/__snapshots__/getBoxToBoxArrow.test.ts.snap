// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBoxToBoxArrow it computes an arrow between boxes 1`] = `
Array [
  16.985053269247754,
  20,
  8.41493643928086,
  20.965910156513193,
  9.380846595794054,
  29.536026986480085,
  1.4585631867350035,
  -0.11223314005989284,
  2.243961350132452,
]
`;

exports[`getBoxToBoxArrow it computes an arrow between boxes with straights disabled 1`] = `
Array [
  16.985053269247754,
  20,
  8.41493643928086,
  20.965910156513193,
  9.380846595794054,
  29.536026986480085,
  1.4585631867350035,
  -0.11223314005989284,
  2.243961350132452,
]
`;

exports[`getBoxToBoxArrow it computes an arrow from a box back to itself 1`] = `
Array [
  15,
  10,
  15,
  0,
  15,
  -10,
  -1.5707963267948966,
  -1.5707963267948966,
  -1.5707963267948966,
]
`;

exports[`getBoxToBoxArrow it computes an arrow from a large box to a small box 1`] = `
Array [
  586,
  308.397718886053,
  468.76455828660926,
  177.92497505063434,
  599.2373021220279,
  60.68953333724373,
  -0.7320098472737407,
  0.8387864795211553,
  -1.5174080106711896,
]
`;
